crates/shared/src/oneinch_api.rs

```diff
@@ -467,12 +468,17 @@ pub struct Protocols {
 #[mockall::automock]
 pub trait OneInchClient: Send + Sync + 'static {
     /// Retrieves a swap for the specified parameters from the 1Inch API.
-    async fn get_swap(&self, query: SwapQuery) -> Result<Swap, OneInchError>;
+    async fn get_swap(
+        &self,
+        query: SwapQuery,
+        set_current_block_header: bool,
```

Comment:
Should this maybe rather be handled by the type implementing `OneInchClient` instead of in the `trait` function signature? If applied to all the API implementation this would probably shrink this PR by 80%.

Reply:
How would that work? This information depends on each request (not necessarily for 1Inch, but e.g. Paraswap a `price` request can either be a quote or native estimate). If we put this on the type implementing the interface we need to switch the flag before invoking get_swap which seems hacky.

Reply:
Or put it inside `SwapQuery` but I guess no big difference anyway...

Reply:
I'm also no a huge fan of flags that control function behaviour. I would, as @MartinquaXD suggests `OneInchClient::new(..., Option<CurrentBlockStream>)` to indicate whether or not the header should be included in the HTTP request. Then, I would build separate 1Inch clients for quoting and price estimation (my rationale being they behave differently, so they should be two different things).

Just a nit though.